starfish/codebook.py

```diff
@@ -442,40 +456,45 @@ def metric_decode(
         Returns
         -------
         IntensityTable :
-            intensity table containing additional data variables for target assignments and metric
-            outputs
+            Intensity table containing normalized intensities, target assignments, distances to
+            the nearest code, and the filtering status of each feature.
 
         """
+
+        self._validate_decode_intensity_input_matches_codebook_shape(intensities)
+
         # normalize both the intensities and the codebook
         norm_intensities, norms = self._normalize_features(intensities, norm_order=norm_order)
         norm_codes, _ = self._normalize_features(self, norm_order=norm_order)
 
-        # mask low intensity features
-        intensity_mask = np.where(norms < min_intensity)[0]
-
         metric_outputs, targets = self._approximate_nearest_code(
             norm_codes, norm_intensities, metric=metric)
 
-        exceeds_distance = np.where(metric_outputs > max_distance)[0]
-        target_index = pd.Index(targets)
+        passes_filters = np.ones_like(targets, dtype=np.bool)
```

Comment:
does it make sense to fill it in one pass?

like `passes_filters = logical_or(norms > min_intensity, metric_outputs < max_distance)`

personally, i feel this makes more sense.  i don't know if it actually works, because i suck at numpy, but I feel it should.